indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -283,6 +288,28 @@ public void onFailure(Throwable throwable)
           waitingForMonitor.wait();
         }
       }
+
+      // Schedule cleanup for task status of the workers that might have disconnected while overlord was not running
+      List<String> workers;
+      try {
+        workers = cf.getChildren().forPath(indexerZkConfig.getStatusPath());
+      }
+      catch (KeeperException.NoNodeException e) {
+        // statusPath doesn't exist yet; can occur if no middleManagers have started.
+        workers = ImmutableList.of();
+      }
+      for (String worker : workers) {
+        if (!zkWorkers.containsKey(worker)
+            && cf.checkExists().forPath(JOINER.join(indexerZkConfig.getAnnouncementsPath(), worker)) == null) {
+          scheduleTasksCleanupForWorker(
+              worker,
+              cf.getChildren()
+                .forPath(JOINER.join(indexerZkConfig.getStatusPath(), worker))
+          );
+        }
+      }
```

Comment:
isn't above code already done in INITIALIZED and is duplicate ? possibly introduced while fixing conflicts ?